livekit-agents/livekit/agents/llm/tool_context.py

```diff
@@ -14,31 +14,36 @@
 
 from __future__ import annotations
 
+import functools
 import inspect
-from abc import ABC
-from collections.abc import Awaitable
+import itertools
+from abc import ABC, abstractmethod
+from collections.abc import Awaitable, Sequence
 from dataclasses import dataclass
 from enum import Flag, auto
-from typing import (
-    Any,
-    Callable,
-    Literal,
-    Protocol,
-    TypeVar,
-    Union,
-    cast,
-    overload,
-    runtime_checkable,
-)
+from typing import Any, Callable, Literal, TypeVar, Union, cast, overload
 
 from typing_extensions import NotRequired, Required, TypedDict, TypeGuard
 
+from . import _provider_format
+
+
+class Tool(ABC):  # noqa: B024
+    pass
+
+
+class ProviderTool(Tool):
+    """Tool that provided by the LLM provider."""
 
-# TODO: refactor Tool inheritance, all tools (FunctionTool, RawFunctionTool, ProviderTool) should inherit from Tool
-class ProviderTool(ABC):  # noqa: B024
     pass
 
 
+class ToolSet(ABC):
```

Comment:
```suggestion
class Toolset(ABC):
```

Reply:
I think `ToolSet` are two words so this capitalization seems more correct

Reply:
fair.. Toolset def seems more "standard".. according to Grok: https://grok.com/share/bGVnYWN5_8f2a4ef7-16be-4cc9-be43-876badd917f4